operator panic after upgrade to v0.16.0 due to missing service httpPort

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]                                                 │
│     panic: runtime error: invalid memory address or nil pointer dereference                                                         │
│ [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1fcdebc]                                                             │
│                                                                                                                                     │
│ goroutine 342 [running]:                                                                                                            │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()                                              │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:115 +0x1fa                             │
│ panic({0x22f1860, 0x400b6f0})                                                                                                       │
│     /usr/local/go/src/runtime/panic.go:884 +0x213                                                                                   │
│ github.com/alexandrevilain/temporal-operator/internal/resource/config.(*ConfigmapBuilder).Update(0xc001b6a430, {0x2c25708?, 0xc0019 │
│     /workspace/internal/resource/config/configmap_builder.go:243 +0x3fc                                                             │
│ github.com/alexandrevilain/controller-tools/pkg/reconciler.(*Reconciler).ReconcileBuilder.func1()                                   │
│     /go/pkg/mod/github.com/alexandrevilain/controller-tools@v0.2.2/pkg/reconciler/reconciler.go:54 +0x2a                            │
│ sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.mutate(0xc001913320?, {{0xc000ddec50?, 0x0?}, {0xc001b5f230?, 0x2c08d9 │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/controller/controllerutil/controllerutil.go:339 +0x4f                    │
│ sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.CreateOrUpdate({0x2c08d98, 0xc000b80f00}, {0x2c15c60, 0xc000154d80}, { │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/controller/controllerutil/controllerutil.go:211 +0x274                   │
│ github.com/alexandrevilain/controller-tools/pkg/reconciler.(*Reconciler).ReconcileBuilder(0xc00097c240, {0x2c08d98, 0xc000b80f00},  │
│     /go/pkg/mod/github.com/alexandrevilain/controller-tools@v0.2.2/pkg/reconciler/reconciler.go:53 +0x115                           │
│ github.com/alexandrevilain/temporal-operator/controllers.(*TemporalClusterReconciler).reconcileResources(0xc00097c280, {0x2c08d98,  │
│     /workspace/controllers/temporalcluster_controller.go:153 +0xe5                                                                  │
│ github.com/alexandrevilain/temporal-operator/controllers.(*TemporalClusterReconciler).Reconcile(0xc00097c280, {0x2c08d98, 0xc000b80 │
│     /workspace/controllers/temporalcluster_controller.go:143 +0x73a                                                                 │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x2c08d98?, {0x2c08d98?, 0xc000b80f00?}, {{{0xc000a8 │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:118 +0xc8                              │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000950aa0, {0x2c08cf0, 0xc00085d450}, {0x2 │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:314 +0x377                             │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000950aa0, {0x2c08cf0, 0xc00085d450})   │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:265 +0x1d9                             │
│ sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()                                                │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:226 +0x85                              │
│ created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2                                         │
│     /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.1/pkg/internal/controller/controller.go:222 +0x587                             │
│ Stream closed EOF for temporal-system/temporal-operator-controller-manager-7884cdc974-wkmbs (manager)                               │
│
```

Downgrading to previously used v0.15.3 resolved the issue. Is there an upgrade step I missed or is this a bug?

--- a/internal/resource/config/configmap_builder.go
+++ b/internal/resource/config/configmap_builder.go
@@ -240,7 +240,6 @@ func (b *ConfigmapBuilder) Update(object client.Object) error {
 				RPC: config.RPC{
 					GRPCPort:        *b.instance.Spec.Services.Frontend.Port,
 					MembershipPort:  *b.instance.Spec.Services.Frontend.MembershipPort,
-					HTTPPort:        *b.instance.Spec.Services.Frontend.HTTPPort,
 					BindOnLocalHost: false,
 					BindOnIP:        "0.0.0.0",
 				},
@@ -289,6 +288,16 @@ func (b *ConfigmapBuilder) Update(object client.Object) error {
 	if !b.instance.Spec.Version.GreaterOrEqual(version.V1_18_0) {
 		temporalCfg.PublicClient = config.PublicClient{
 			HostPort: b.instance.GetPublicClientAddress(),
+		}
+	}
+
+	// Temporal >= 1.22 provides HTTP endpoint for the frontend
+	if b.instance.Spec.Version.GreaterOrEqual(version.V1_22_0) &&
+		b.instance.Spec.Services.Frontend.HTTPPort != nil {
+		frontend, ok := temporalCfg.Services[string(primitives.FrontendService)]
+		if ok {
+			frontend.RPC.HTTPPort = *b.instance.Spec.Services.Frontend.HTTPPort
+			temporalCfg.Services[string(primitives.FrontendService)] = frontend
 		}
 	}
 

--- a/pkg/version/version.go
+++ b/pkg/version/version.go
@@ -38,6 +38,7 @@ var (
 	V1_18_0 = MustNewVersionFromString("1.18.0") //nolint:stylecheck,revive
 	V1_20_0 = MustNewVersionFromString("1.20.0") //nolint:stylecheck,revive
 	V1_21_0 = MustNewVersionFromString("1.21.0") //nolint:stylecheck,revive
+	V1_22_0 = MustNewVersionFromString("1.22.0") //nolint:stylecheck,revive
 )
 
 // Version is a wrapper around semver.Version which supports correct

